### PR TITLE
bayescan: init at 2.1

### PR DESCRIPTION
**Pull request by bzizou**
###### Motivation for this change
Tool used into my computing center

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/applications/science/biology/bayescan/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  installPhase = ''
+    mkdir -p $out/bin
+    mkdir -p $out/share/doc/bayescan
+    cp bayescan_${version} $out/bin
+    cp -r ../*pdf ../input_examples ../"R functions" $out/share/doc/bayescan
+  '';
+
+  meta = {
+    description = "Detecting natural selection from population-based genetic data";
+    license     = "GPL2";
```
Comment (risicle) [inline pkgs/applications/science/biology/bayescan/default.nix]: This should be one of the license attrsets specified in https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix - see https://nixos.org/nixpkgs/manual/#sec-meta-license

File: pkgs/applications/science/biology/bayescan/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  };
+
+  nativeBuildInputs = [ unzip ];
+
+  # Disable FORTIFY_SOURCE or the binary fails with "buffer overflow"
+  hardeningDisable = [ "fortify" ];
+
+  sourceRoot = "BayeScan${version}/source";
+
+  preBuild = ''
```
Comment (risicle) [inline pkgs/applications/science/biology/bayescan/default.nix]: We tend to do such things as `postPatch` rather than `preBuild` :man_shrugging:

**Comment by risicle** [comment]
To get this building on macos (possibly all llvm platforms), you just need to... add another replacement for the `Makefile`:

```
substituteInPlace Makefile --replace "-static" "" \
  --replace "g++" "c++"
```

and provide it with `llvmPackages.openmp`:

```
  buildInputs = stdenv.lib.optional stdenv.cc.isClang llvmPackages.openmp;
```

Given this it builds for me, macos 10.13 & non-nixos linux x86_64.

**Comment by bzizou** [comment]
up?

**Comment by veprbl** [comment]
@GrahamcOfBorg build bayescan

**Pull request merged**